### ISSUE TITLE
[no-identical-title] doesn't handle tests within functions correctly

### ISSUE THREAD

**Issue by MarcellPerger1** [jest-community/eslint-plugin-jest#1305]
This
```js
function test_x_1() {
  it("Tests x", () => {
    expect("x").toBe("x");
  })
}

function test_x_2() {
  it("Tests x", () => {
    expect("x2").toBe("x2");
  })
}

describe("Test 1", () => {
  // (other tests)
  test_x_1();
})
describe("Test 2", () => {
  // (other tests)
  test_x_2();
})
```
gives the error
```
/home/runner/eslintjestbug1/index.test.js
  8:6  error  Test title is used multiple times in the same describe block  jest/no-identical-title

✖ 1 problem (1 error, 0 warnings)
```
when no errors are expected since the functions are called from within 2 different `describe` blocks.

**Comment by G-Rath** [comment]
I'm going to close this as while technically it's not correct, it's not easy for us to accurately track this and I think semantically it's better - the idea of the rule is that having the same full title can be confusing, and there's nothing stopping someone from calling both functions in the same `describe`.

Personally I'd be scared of the example you've given since it's two different tests somehow being described exactly the same; I know it's not a real-world example but that actually doesn't help it since I can't think of real-world cases where this would happen (the examples I can think of either can have different titles or are generic so use the same function).

Happy to reconsider if we get more feedback but so far this is the only questioning of this behaviour in the 3+ years the rule has been around.
